### ISSUE TITLE
Enhancement: [prefer-enum-initializers] enforce first value only for ordinals

### ISSUE THREAD

**Issue by mistic100** [typescript-eslint/typescript-eslint#10628]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-enum-initializers/

### Description

I sometimes use ordinal enums for technical states, because I think it easier to read than string enums (I still always use string enums for functionnal states, mainly because the string representation is expected by the backend).

The actual value of these enums is not important, it is not shared accross mutliple packages.

But there is one mistake that can occur easily: check the presence of a variable with a simple `if` statement, which will fail on the first enum value (defaults to 0).

Example use case:

```ts
enum GridAction {
  CLICK,
  RIGHT_CLICK,
  DOUBLE_CLICK
}

doSomething(action?: GridAction) {
  if (action) {
    // will not pass if action = CLICK
  }
}
```

This is easily fixable by defining the value of the first value to 1

```ts
enum GridAction {
  CLICK = 1,
  RIGHT_CLICK,
  DOUBLE_CLICK
}
```

So this request is to add an option to the `prefer-enum-initializers` to allow, for ordinals enums, to only define the first value. It would not enforce the value, but the cognitive effort to define it should be enough to choose to define it to 1 if needed.

### Fail

```TypeScript
enum GridAction {
  CLICK,
  RIGHT_CLICK,
  DOUBLE_CLICK
}

enum GridAction {
  CLICK = 'CLICK',
  RIGHT_CLICK,
  DOUBLE_CLICK
}
```

### Pass

```TypeScript
enum GridAction {
  CLICK = 1,
  RIGHT_CLICK,
  DOUBLE_CLICK
}

enum GridAction {
  CLICK = 0,
  RIGHT_CLICK = 1,
  DOUBLE_CLICK = 2
}

enum GridAction {
  CLICK = 'CLICK',
  RIGHT_CLICK = 'RIGHT_CLICK',
  DOUBLE_CLICK = 'DOUBLE_CLICK'
}
```

**Comment by kirkwaiblinger** [comment]
Hey @mistic100 ! 

I understand the request but I'm not super sure we'll want to support it... We generally discourage making assumptions about the enum constituents' runtime values. There's never a need to use truthiness with enums, either, since you could just as well write 
```ts
if (action != null) {
}
```
If you do write `if (action)` it forces any reviewer of your code to do a double-take to see whether you actually _meant_ to omit an enum constituent with value 0 (or '') from the check, even if most of your enums are initialized to truthy values. This double-take is completely avoidable by just not relying on the enum's truthiness, which you can enforce with strict-boolean-expressions. Therefore, my impression for now is that I personally lean against this request.

See also #7149, where this came up a bit as well.

@typescript-eslint/triage-team

**Comment by mistic100** [comment]
Okay but then is there a rule that could prevent to write `if (action)` and enforce `if (action != null)` (not sure this pass "eqeqeq") or using an helper like lodash `if (!isNil(action))` ?

**Comment by kirkwaiblinger** [comment]
> which you can enforce with strict-boolean-expressions

docs: https://typescript-eslint.io/rules/strict-boolean-expressions/
([playground](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=KYOwrgtgBA4gTgSwCYEEDGAXBB7EUDeAUFFAMIAyAkqQNIA0xUASpTABIAqA%2BhdfYwBEA8gFUAQuQCiPKrUIBfQoQBmYEJhx4k2AMrYIwDAAsEIAOYAKAIYbcAfgBcsRKlsgAlAUYJlUa288iEhIAehCoAHcEABtoqBBsDCgABysAZzSoHygbLFwoAF4yWRpGRXkgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZfKsugIwHs%2BSAIZN6AD2LQUySnyaoMkAO5DoTSODABfEFqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))

**Comment by kirkwaiblinger** [comment]
regarding eqeqeq: yeah it's entirely up to you on the `action != null` (the standard inside our repo, and my personal recommendation) or strict equality `action !== undefined`/`action !== null` (which expends extra brain cells thinking about _which_ type of nullishness you're checking for).

**Comment by mistic100** [comment]
Sorry missed that sentence!

I'll try that by only keeping `allowNullableEnum: false` (for now, I cannot push breaking changes easily in our enterprise tooling)

**Comment by kirkwaiblinger** [comment]
For sure, you may be up against a classic "the best time to lint your code was before it was written" situation 🙃
